### PR TITLE
Avoiding translation calls for the default locale

### PR DESCRIPTION
**Pull request by Vitalij-Voronkoff**
BroadleafCommerce/QA#3782

### CODE REVIEW

File: common/src/main/java/org/broadleafcommerce/common/i18n/service/DynamicTranslationProvider.java
Comment (ckittrell) [inline common/src/main/java/org/broadleafcommerce/common/i18n/service/DynamicTranslationProvider.java]:
```suggestion
     * If in the context of the default locale, then skip all translation calls and return the defaultValue.
```

**Review by ckittrell — APPROVED** [review]
Approving with the expectation that we cleanup the javadoc message

File: common/src/main/java/org/broadleafcommerce/common/i18n/service/DynamicTranslationProvider.java
Comment (ckittrell) [inline common/src/main/java/org/broadleafcommerce/common/i18n/service/DynamicTranslationProvider.java]: @Vitalij-Voronkoff On second thought, I think we should refactor this to be `&& localeShouldBeTranslated(requestContext.getLocale())`

Within that method, we can simply return `!locale.getDefaultFlag()`, but it would give users the ability to declare that `en_us` & `en_gb` should not be translated if only `en_us` is the default.

**Review by ckittrell — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/broadleafcommerce/common/i18n/service/DynamicTranslationProvider.java
Comment (ckittrell) [reply common/src/main/java/org/broadleafcommerce/common/i18n/service/DynamicTranslationProvider.java]: ... realizing that `getValue(...)` is a static method which really derails the idea of extensibility

**Review by ckittrell — APPROVED** [review]

**Comment by ckittrell** [comment]
@danielcolgrove @jfleschler I made a fair amount of changes on this one, so I'd like a second pair of eyes on it from one of you guys.

File: common/src/main/java/org/broadleafcommerce/common/i18n/service/TranslationServiceImpl.java
Comment (jfleschler) [inline common/src/main/java/org/broadleafcommerce/common/i18n/service/TranslationServiceImpl.java]: This is an API change that could break clients.  We probably don't want to have this as part of the PR

**Review by jfleschler — CHANGES_REQUESTED** [review]

**Review by jfleschler — APPROVED** [review]
LGTM

**Pull request merged**
